### PR TITLE
feat(email-verification): added method in userService to send email verification link to a user

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes - https://github.com/fabric8-services/fabric8-auth/issues/455

[Wireframe](https://redhat.invisionapp.com/share/E9FH915D3U7#/screens/285588330) for the design where the link will be there if a user's email is not verified. An option for the user to resend verification link to his email.

### CODE REVIEW

File: src/app/user/user.service.ts [hunk 1]
```diff
@@ -185,6 +185,17 @@ export class UserService {
       })
   }
 
+  /**
+   * Send email verification link to user.
+   */
+  sendEmailVerificationLink(): Observable<any> {
```
Comment (jiekang) [inline src/app/user/user.service.ts]: Please return the actual type, not `any`.

File: src/app/user/user.service.ts [hunk 2]
```diff
@@ -185,6 +185,17 @@ export class UserService {
...
+    return this.http
+      .post(this.usersUrl + '/verificationcode', null, { headers: this.headers })
+      .map(response => {
```
Comment (jiekang) [inline src/app/user/user.service.ts]: Please annotate response with it's correct type. E.g. `(response: Response)` if it's a Response.

File: src/app/user/user.service.ts [hunk 3]
```diff
@@ -185,6 +185,17 @@ export class UserService {
       })
   }
 
+  /**
+   * Send email verification link to user.
+   */
+  sendEmailVerificationLink(): Observable<any> {
+    return this.http
+      .post(this.usersUrl + '/verificationcode', null, { headers: this.headers })
```
Comment (jiekang) [inline src/app/user/user.service.ts]: Prefer empty string '' instead of null body.

**Comment by joshuawilson** [comment]
@jiekang do you approve?

**Pull request merged**